SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialGameState.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGameState.h"
+#include "Net/UnrealNetwork.h"
+#include "SpatialGDKSettings.h"
+#include "SpatialMetricsDisplay.h"
+
+ASpatialGameState::ASpatialGameState() :
```

Comment:
I don't think we should add a new SpatialGameState, we want to avoid injecting new classes into the middle of existing Unreal hierarchies (I realise we already have SpatialGameInstance, but we're looking at incorporating that functionality directly). We don't necessarily need anything to keep a reference to ASpatialMetricsDisplay directly either, and can ask the client to search for the object class type when needed.

Reply:
Yea just trying to reduce barrier to entry for customer teams.  The alternative you're thinking is to have them just drop one of the ASpatailMetricsDisplay into each level they want the stats for?

Reply:
What if we move it into `USpatialNetDriver::CreateAndInitializeCoreClasses`?

Reply:
I tried moving it to the suggested location but it doesn't map well there (console commands not supported in that class without extra plumbing, we'll get one instance per connection of the SpatialMetricsDisplay rather than one for the deployment, etc).

I'll remove SpatialGameState and push the supporting code for SpatialMetricsDisplay down into the NWX-specific GameState.  I'll also add documentation to describe how other teams can do the same.